crates/ty_python_semantic/src/semantic_index/use_def.rs

```diff
@@ -1,12 +1,22 @@
 //! First, some terminology:
 //!
-//! * A "binding" gives a new value to a variable. This includes many different Python statements
+//! * A "place" is semantically a location where a value can be read or written, and syntactically,
+//!   an expression that can be the target of an assignment, e.g. `x`, `x[0]`, `x.y`. (The term is
+//!   borrowed from Rust). In Python syntax, an expression like `f().x` is also allowed as the
+//!   target so it can be called a place, but we do not record declarations / bindings like `f().x:
+//!   int`, `f().x = ...`. Type checking itself can be done by recording only assignments to names,
+//!   but in order to perform type narrowing by attribute/subscript assignments, they must also be
+//!   recorded. We also record assignments to invalid places since whether a place expression is
+//!   valid is determined during type inference, but invalid assignments do not affect type
+//!   narrowing.
```

Comment:
What do you mean by an "invalid place" here? Because from what I see in the code, we only record assignments to valid places (that is, those which can be parsed into a `PlaceExpr`).